week4/find-your-github/src/pages/search.tsx

```diff
@@ -47,21 +48,59 @@ const InputButton = styled.button`
   cursor: pointer;
 `;
 
+const NoUserBox = styled.div`
+  background-color: ${({ theme }) => theme.colors.subColor};
+  color: ${({ theme }) => theme.colors.buttonColor};
+  font-size: ${({ theme }) => theme.fontSizes.text};
+  width: 13rem;
+  height: 10rem;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: column;
+  border-radius: 6px;
+`;
+
+export interface userProfileProps {
+  name: string;
+  avatar_url: string;
+  gitHubUrl: string;
+  following: number;
+  followers: number;
+  public_repos: number;
+}
+
 export default function Search() {
+  const [userProfile, setUserProfile] = useState<userProfileProps>();
   const [userName, setUserName] = useState("");
   const navigate = useNavigate();
+  const [isExist, setIsExist] = useState<boolean>(false);
 
   const onUserNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const value = e.target.value;
     setUserName(value);
   };
 
   const clickSearchButton = async () => {
-    if (!userName) return;
     const data = await getGitHubProfile(userName);
-    console.log(data);
-    setUserName("");
-    navigate(`/search/${userName}`);
+    if (data === "noUser") {
+      setIsExist(false);
+    } else {
+      console.log(data);
+      setUserProfile(data);
+      setUserProfile({
+        ...data,
+        name: data.login,
+        avatar_url: data.avatar_url,
+        gitHubUrl: data.html_url,
+        following: data.following,
+        followers: data.followers,
+        public_repos: data.public_repos,
```

Comment:
이고 
```suggestion 
setUserProfile({
        ... userProfile,
        data
          });
```

이케해도 또가튼거 아니에여?!

Reply:
```js
      setUserProfile({
        ...userProfile,
        login: data.login,
        name: data.name,
        avatar_url: data.avatar_url,
        gitHubUrl: data.html_url,
        following: data.following,
        followers: data.followers,
        public_repos: data.public_repos,
      });
```
이렇게 바꿔야하는거 아니야?! data에 요소가 많은데 어떤 요소가 login, name ... 에 매치될지 모르니까??

Reply:
아 data 한번 타입 걸러서오능줄알아또!!